pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -131,6 +135,24 @@ func (r *pullthroughBlobStore) ServeBlob(ctx context.Context, w http.ResponseWri
 	return nil
 }
 
+// Create needs to be overridden in order to use pullthroughBlobStore middleware during cross-repository push so that remote layers don't need to be pushed.
+func (r *pullthroughBlobStore) Create(ctx context.Context, options ...distribution.BlobCreateOption) (distribution.BlobWriter, error) {
+	options = append(options, storage.WithRepositoryMiddlewareWrapper(
+		func(ctx context.Context, repo distribution.Repository, name reference.Named) (distribution.Repository, error) {
+			nameParts := strings.SplitN(name.Name(), "/", 2)
```

Comment:
We're doing it the same way on most other places. And AFAIK upstream registry doesn't support more than 2 segments either. We could create a utility function to do this. But I wouldn't clobber this with such a refactor.